### ISSUE TITLE
mlaunch fails to launch simulators if the Simulator app is already running

### ISSUE THREAD

**Issue by rpendleton** [xamarin/xamarin-macios#14560]
### Steps to Reproduce

1. Launch the simulator app
2. Close any running simulators, but don't close the Simulator app entirely
3. Do something that relies on mlaunch to start a simulator. For example:
   ```
   mlaunch --launchsim path/to/AnyApp.app --device=:v2:udid=ED2EA668-2181-490E-8F32-042E7A0D316F
   ```

### Expected Behavior

I expect the requested simulator to boot.

### Actual Behavior

The Simulator app is brought to the foreground, but the device doesn't boot. The mlaunch command exits with a status code of 1 and displays the following error when verbose logs are enabled.

(If you completely quit the Simulator app, or if you manually boot the desired simulator, everything works as expected.)

```
Using Xcode 13.3 found in /Applications/Xcode.app/Contents/Developer
Xamarin.Hosting: Xamarin.Hosting
Xamarin.Hosting:     Version: 696db0a953 (main)
Xamarin.Hosting:     Xcode: /Applications/Xcode.app
Xamarin.Hosting:     Xcode Version: 13.3
Xamarin.Hosting:     Verbosity: 4
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTFoundation.framework/Versions/A/DVTFoundation
Xamarin.Hosting: Loaded /Library/Developer/PrivateFrameworks/CoreSimulator.framework/CoreSimulator
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTDeviceFoundation.framework/DVTDeviceFoundation
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DTDeviceKitBase.framework/DTDeviceKitBase
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTKit.framework/DVTKit
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DTDeviceKit.framework/DTDeviceKit
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DTXConnectionServices.framework/DTXConnectionServices
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTSourceControl.framework/DVTSourceControl
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTServices.framework/DVTServices
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTPortal.framework/DVTPortal
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTDocumentation.framework/DVTDocumentation
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTAnalyticsClient.framework/DVTAnalyticsClient
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/XCServices.framework/XCServices
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTAppStoreConnect.framework/DVTAppStoreConnect
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/ContentDeliveryServices
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTITunesSoftwareServiceFoundation.framework/DVTITunesSoftwareServiceFoundation
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTITunesSoftware.framework/DVTITunesSoftware
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTProducts.framework/DVTProducts
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTProductsUI.framework/DVTProductsUI
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/DVTAnalytics.framework/DVTAnalytics
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/SourceKit.framework/SourceKit
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/Frameworks/IDEFoundation.framework/IDEFoundation
Xamarin.Hosting: Loaded /Applications/Xcode.app/Contents/SharedFrameworks/IDEProducts.framework/IDEProducts
Xamarin.Hosting: Simulator watchdogs are already disabled for 'iOS 15.0 (19A339) - iPhone Xʀ'.
Xamarin.Hosting: Launching simulator application 'com.apple.iphonesimulator'
error HE0018: Could not launch the simulator application: The application “Simulator” could not be launched because a miscellaneous error occurred.

  at Xamarin.Hosting.SimulatorApplication.LaunchAsync (Xamarin.Hosting.Bindings.SimDevice device) [0x0020d] in /Users/builder/azdo/_work/2/s/maccore/tools/mlaunch/Xamarin.Hosting/Xamarin.Hosting/SimulatorApplication.cs:203
  at Xamarin.Hosting.Services.LaunchSimulatorAsync (Xamarin.Hosting.Bindings.SimDevice device) [0x000a7] in /Users/builder/azdo/_work/2/s/maccore/tools/mlaunch/Xamarin.Hosting/Xamarin.Hosting/Services.cs:1403
  at Xamarin.Launcher.Driver.LaunchSimulatorAsync () [0x00150] in /Users/builder/azdo/_work/2/s/maccore/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/Main.cs:782
  at Xamarin.Utils.NSRunLoopExtensions.RunUntilTaskCompletion[T] (Foundation.NSRunLoop this, System.Threading.Tasks.Task`1[TResult] task) [0x00082] in /Users/builder/azdo/_work/2/s/maccore/tools/mlaunch/Xamarin.Hosting/Xamarin.Utils/Extensions.cs:35
  at Xamarin.Launcher.Driver.Main2 (System.String[] args) [0x00b43] in /Users/builder/azdo/_work/2/s/maccore/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/Main.cs:459
  at Xamarin.Launcher.Driver.Main (System.String[] args) [0x0006d] in /Users/builder/azdo/_work/2/s/maccore/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/Main.cs:151
```

### Environment

```
=== Visual Studio Professional 2019 for Mac ===

Version 8.10.21 (build 4)
Installation UUID: c1917f5f-db54-4b43-a8d9-0ffbe14fdf32
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000162

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.162 (2020-02/2ca650f1f62) (64-bit)
	Package version: 612000162

=== Roslyn (Language Service) ===

3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb

=== NuGet ===

Version: 5.9.0.7134

=== .NET SDK (x64) ===

SDK: /usr/local/share/dotnet/sdk/6.0.201/Sdks
SDK Versions:
	6.0.201
	6.0.103
	6.0.101
	5.0.406
	5.0.404
	5.0.403
	3.1.417
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	6.0.3
	6.0.1
	5.0.15
	5.0.13
	5.0.12
	3.1.23
	3.1.22
	2.1.23

=== .NET Core 3.1 SDK ===

SDK: 3.1.417

=== .NET 5.0 SDK ===

SDK: 5.0.406

=== Xamarin.Profiler ===

Version: 1.6.15.68
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 13.3 (20102)
Build 13E113

=== Xamarin.Mac ===

Version: 8.8.0.0 (Visual Studio Professional)
Hash: f10d9e023
Branch: xcode13.3
Build date: 2022-03-15 11:47:34-0400

=== Xamarin.iOS ===

Version: 15.8.0.0 (Visual Studio Professional)
Hash: f10d9e023
Branch: xcode13.3
Build date: 2022-03-15 11:47:35-0400

=== Xamarin Designer ===

Version: 16.11.0.60
Hash: 56f9b80b0
Branch: remotes/origin/d16-11
Build date: 2021-12-15 01:44:16 UTC

=== Xamarin.Android ===

Version: 12.0.0.3 (Visual Studio Professional)
Commit: xamarin-android/d16-11/f0e3c2d
Android SDK: /Users/i21921/Library/Android/sdk
	Supported Android versions:
		None installed

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 33.0.1
SDK Build Tools Version: 33.0.0 rc2

Build Information: 
Mono: c633fe9
Java.Interop: xamarin/java.interop/d16-11@476bb5b
ProGuard: Guardsquare/proguard/v7.0.1@912d149
SQLite: xamarin/sqlite/3.35.4@85460d3
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-11@87af37b

=== Eclipse Temurin JDK ===

Java SDK: Not Found

=== Android SDK Manager ===

Version: 16.10.0.13
Hash: 1b81df5
Branch: remotes/origin/d16-10
Build date: 2021-11-12 00:17:32 UTC

=== Android Device Manager ===

Version: 16.10.0.15
Hash: 89dcc0b
Branch: remotes/origin/d16-10
Build date: 2021-11-12 00:17:52 UTC

=== Build Information ===

Release ID: 810210004
Git revision: eb0b2f7259d35b7d767c79d91d356881227e0985
Build date: 2022-03-17 17:18:28-04
Build branch: release-8.10

=== Operating System ===

Mac OS X 12.3.0
Darwin 21.4.0 Darwin Kernel Version 21.4.0
    Mon Feb 21 20:34:37 PST 2022
    root:xnu-8020.101.4~2/RELEASE_X86_64 x86_64
```
AB#1509963

**Comment by rolfbjarne** [comment]
I can reproduce this.

**Comment by IoTFier** [comment]
Ditto. Same problem. After XCode upgrade to 13.3, I can't start the simulators to. It comes up with black screen.

**Comment by rolfbjarne** [comment]
> Just in case this is useful information, I installed the CI packages from #14576 and can confirm they fix the issue for me.

Thanks for confirming!

**Comment by drmaven** [comment]
> Just in case this is useful information, I installed the CI packages from #14576 and can confirm they fix the issue for me.

Hi, could you share some detailed instructions on how to install the fix?

**Comment by rpendleton** [comment]
> Hi, could you share some detailed instructions on how to install the fix?

@drmaven

f2005bab4d is the commit that backported the fix to the Xcode 13.3 branch. If you click that commit hash, you can open the build checks for the commit by clicking the red X or green checkmark on the left-side of the commit message. (From what I've seen, most of the commits in xamarin-macios seem to fail at least some of the build checks, so it tends to be the red X.)

Once you click the red X to open the build checks, one of the entries will be "PKG-Xamarin.iOS-notarized", and you can download and install that by clicking the "Details" link.

Alternatively, you can download it directly using this link:

https://bosstoragemirror.blob.core.windows.net/wrench/xcode13.3/f2005bab4dcccfdb139c1fc62285e19abbbc5580/5962973/package/notarized/xamarin.ios-15.8.0.4.pkg

**Comment by drmaven** [comment]
> > Hi, could you share some detailed instructions on how to install the fix?
> 
> @drmaven
> 
> [f2005ba](https://github.com/xamarin/xamarin-macios/commit/f2005bab4dcccfdb139c1fc62285e19abbbc5580) is the commit that backported the fix to the Xcode 13.3 branch. If you click that commit hash, you can open the build checks for the commit by clicking the red X or green checkmark on the left-side of the commit message. (From what I've seen, most of the commits in xamarin-macios seem to fail at least some of the build checks, so it tends to be the red X.)
> 
> Once you click the red X to open the build checks, one of the entries will be "PKG-Xamarin.iOS-notarized", and you can download and install that by clicking the "Details" link.
> 
> Alternatively, you can download it directly using this link:
> 
> https://bosstoragemirror.blob.core.windows.net/wrench/xcode13.3/f2005bab4dcccfdb139c1fc62285e19abbbc5580/5962973/package/notarized/xamarin.ios-15.8.0.4.pkg

Thank you so much for the detailed instructions, much appreciated. 

Unfortunately, the issue still occurs even when using the updated SDK.

**Comment by rolfbjarne** [comment]
@drmaven please [file a new issue](https://github.com/xamarin/xamarin-macios/issues/new) and attach the logs that shows the error you're seeing, and we'll have a look.
